### ISSUE TITLE
exclusive/inclusive lighting

### ISSUE THREAD

**Issue by lacilaci** [LuxCoreRender/LuxCore#147]
This should be light feature where we could use lists of included or excluded objects for a particular light.
With corona for blender addon this is controlled through nodes where you can create object list node and plug it into exclude or include input on light emmision node.

It is a very useful feature for tweaking scene look in-camera, so that for example if you have objects that you want to make pop out or objects that due to natural lighting in a particular scene appear flat, you can create "fake" invisible light source affecting only that object.

**Comment by thiagodesul** [comment]
Any news on this? Light-Linking is a very necessary feature for Lighting artists!

**Comment by Dade916** [comment]
This kind of features are usually based on the assumption that light is sampled in a single way. This is not true in LuxCore where MIS (multiple importance sampling is used) is widely used.

Aside from the general consideration, I could implemented light-linking for direct light sampling however it is unclear to me what is supposed to be the behavior with indirect light.

For instance, an object can receive the light, of an excluded light source, bouncing over another surface: should it be enabled or not ?

The problem can easily become unmanageable after few bounces so I assume you are talking of direct light only (?).

**Comment by lacilaci** [comment]
wait a second. If we exclude light tracing and BiDir. Is this doable? Cause there are a lot of features not working for all integrators, so a "pathOCL" coul have this light linking feature while BiDir won't...? 
This is not a super essential feature but even in outdated corona plugin for blender you can create list of objects that are/aren't affected by light in node editor... So, maybe a pathOCL exclusive feature. Or once luxcore is running on RTX?

**Comment by Dade916** [comment]
Yes, never thought to implement this in BiDir, it would be a PATHOCL/CPU only feature but it still use MIS so the light received by a light source is still the result of two things:

1) direct light sampling (here is easy to include, exclude lights ... caches aside);

2) a path randomly bouncing around and, may be, hitting one of the light sources. What I'm supposed to do with that contribution ? Use it, ignore it ? Track what surfaces has been hit and their respective include/exclude lists ... this can easily become a nightmare after a couple of bounces.
